v2/cmd/asoctl/pkg/importresources/importable_arm_resource.go

```diff
@@ -442,9 +445,10 @@ func (i *importableARMResource) getStatus(
 	ctx context.Context,
 	armID string,
 	importable genruntime.ImportableARMResource,
+	factory *importFactory,
```

Comment:
We can pass _just the scheme_ here rather than the whole struct?

Reply:
I have plans to tidy this further in the future, adding new methods to the factory that simplify this code further.